servers/rest/rest.go

```diff
@@ -38,7 +38,7 @@ func NewMuxed(m cmux.CMux) *GDRest {
 
 // Serve begins serving client HTTP requests served by REST server
 func (r *GDRest) Serve() {
-	chain := alice.New(middleware.LogRequest, middleware.ReqIDGenerator).Then(r.Routes)
+	chain := alice.New(middleware.Auth, middleware.LogRequest, middleware.ReqIDGenerator).Then(r.Routes)
```

Comment:
The `LogRequest` should come at first because we want all request and response to be logged, including 401/403

```
INFO[2017-06-08T07:45:48+05:30] 127.0.0.1 - - [08/Jun/2017:07:45:48 +0530] "GET /v1/peers HTTP/1.1" 200 131 
```